sympy/solvers/ode/single.py

```diff
@@ -1889,5 +1889,87 @@ def _get_general_solution(self, *, simplify_flag: bool = True):
         return fsol
 
 
+class Hypergeometric2nd(SingleODESolver):
+    r"""
+    Solves 2nd order linear differential equations.
+
+    It computes special function solutions which can be expressed using the
+    2F1, 1F1 or 0F1 hypergeometric functions.
+
+    .. math:: y'' + A(x) y' + B(x) y = 0\text{,}
+
+    where `A'/2 + A^2/4 - B`  is a rational function of x.
+
+    These kinds of differential equations have solution of non-Liouvillian form.
+
+    Given linear ODE can be obtained from 2F1 given by
+
+    .. math:: (x^2 - x) y'' + ((a + b + 1) x - c) y' + b a y = 0\text{,}
+
+    where {a, b, c} are arbitrary constants.
+
+    A note: Although the paper in reference describes an algorithm that should find all solutions
+    of the form of below equation for 2F1, 0F1 and 1F1 only the 2F1 case is implemented in
+    Sympy right now.
+
+    .. math:: y = P(x) _pF_q(..; ..;\frac{\alpha x^k + \beta}{\gamma x^k + \delta})\text{,}
+
+    where pFq is any of 2F1, 1F1 or 0F1.
```

Comment:
This should go in a
```
Notes
=====
```
section and should be worded better. The equation should be part of a sentence.